File: GuideObjC/GuideObjC/Extensions/EGFHumanName+Additions.m
//
//  EGFHumanName+Additions.m
//  GuideObjC
//
//  Created by LuzanovRoman on 14.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "EGFHumanName+Additions.h"

@implementation EGFHumanName (Additions)

- (NSString *)fullName {
    NSString * fullName = @"";
    
    if (self.family) {
        fullName = self.family;
    }
    if (self.given) {
        fullName = fullName.length == 0 ? self.given : [fullName stringByAppendingFormat:@" %@", self.given];
    }
    return fullName;
}
@end

File: GuideSwift/GuideSwift/Extensions/EGFHumanName+Additions.swift
//
//  EGFHumanName+Additions.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 13.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import Foundation

extension EGFHumanName {

    func fullName() -> String? {
        var fullName = ""

        if let value = family {
            fullName = value
        }
        if let value = given {
            fullName += fullName.isEmpty ? value : " " + value
        }
        return fullName.isEmpty ? nil : fullName
    }
}
